src/equations/ideal_glm_mhd_multiion_2d.jl

```diff
@@ -482,6 +535,111 @@ The term is composed of four individual non-conservative terms:
+    @unpack charge_to_mass = equations
+    # Unpack left and right states to get the magnetic field
+    B1_ll, B2_ll, B3_ll = magnetic_field(u_ll, equations)
+    B1_rr, B2_rr, B3_rr = magnetic_field(u_rr, equations)
+    psi_ll = divergence_cleaning_field(u_ll, equations)
+    psi_rr = divergence_cleaning_field(u_rr, equations)
+
+    # Compute important averages
+    B1_avg = 0.5f0 * (B1_ll + B1_rr)
+    B2_avg = 0.5f0 * (B2_ll + B2_rr)
+    B3_avg = 0.5f0 * (B3_ll + B3_rr)
+    mag_norm_ll = B1_ll^2 + B2_ll^2 + B3_ll^2
+    mag_norm_rr = B1_rr^2 + B2_rr^2 + B3_rr^2
+    mag_norm_avg = 0.5f0 * (mag_norm_ll + mag_norm_rr)
+    psi_avg = 0.5f0 * (psi_ll + psi_rr)
+
+    # Mean electron pressure
+    pe_ll = equations.electron_pressure(u_ll, equations)
+    pe_rr = equations.electron_pressure(u_rr, equations)
+    pe_mean = 0.5f0 * (pe_ll + pe_rr)
+
+    # Compute charge ratio of u_ll
+    charge_ratio_ll = zero(MVector{ncomponents(equations), eltype(u_ll)})
+    total_electron_charge = zero(eltype(u_ll))
+    for k in eachcomponent(equations)
+        rho_k = u_ll[3 + (k - 1) * 5 + 1]
+        charge_ratio_ll[k] = rho_k * charge_to_mass[k]
+        total_electron_charge += charge_ratio_ll[k]
+    end
+    charge_ratio_ll ./= total_electron_charge
+
+    # Compute auxiliary variables
+    v1_plus_ll, v2_plus_ll, v3_plus_ll, vk1_plus_ll, vk2_plus_ll, vk3_plus_ll = charge_averaged_velocities(u_ll,
+                                                                                                           equations)
+    v1_plus_rr, v2_plus_rr, v3_plus_rr, vk1_plus_rr, vk2_plus_rr, vk3_plus_rr = charge_averaged_velocities(u_rr,
+                                                                                                           equations)
+
+    f = zero(MVector{nvariables(equations), eltype(u_ll)})
+
+    # Entries of Godunov-Powell term for induction equation (multiply by 2 because the non-conservative flux is
+    # multiplied by 0.5 whenever it's used in the Trixi code)
+    v_plus_ll_normal = v1_plus_ll * normal_direction[1] +
+                       v2_plus_ll * normal_direction[2]
+    f[1] = 2 * v_plus_ll_normal * B1_avg
+    f[2] = 2 * v_plus_ll_normal * B2_avg
+    f[3] = 2 * v_plus_ll_normal * B3_avg
+
+    for k in eachcomponent(equations)
+        # Compute term Lorentz term
+        f2 = charge_ratio_ll[k] *
+             ((0.5f0 * mag_norm_avg - B1_avg * B1_avg + pe_mean) * normal_direction[1] +
+              (-B2_avg * B1_avg) * normal_direction[2])
+        f3 = charge_ratio_ll[k] * ((-B1_avg * B2_avg) * normal_direction[1] +
+              (-B2_avg * B2_avg + 0.5f0 * mag_norm_avg + pe_mean) * normal_direction[2])
+        f4 = charge_ratio_ll[k] * (-B1_avg * B3_avg * normal_direction[1] -
+              B2_avg * B3_avg * normal_direction[2])
+        f5 = (vk1_plus_ll[k] * normal_direction[1] +
+              vk2_plus_ll[k] * normal_direction[2]) * pe_mean
```

Comment:
I had already suggested changing this in https://github.com/trixi-framework/Trixi.jl/pull/2446#discussion_r2161445654.

This would be more efficient if you computed `B_dot_n_avg` at the beginning of the function, as in the 3D version. This quantity is the same for all species.